plugin/aws/s3Bucket.go

```diff
@@ -186,6 +196,21 @@ func (b *s3Bucket) List(ctx context.Context) ([]plugin.Entry, error) {
 	return listObjects(ctx, b.client, b.Name(), "")
 }
 
+func (b *s3Bucket) Delete(ctx context.Context) (bool, error) {
+	// According to https://docs.aws.amazon.com/AmazonS3/latest/dev/delete-or-empty-bucket.html,
+	// we must delete the bucket's objects and object versions (for versioned buckets) before
+	// deleting the bucket itself. We defer deleting object versions as part of supporting
+	// versioned buckets, which is a separate piece of work.
```

Comment:
Should this be an issue, or should we wait until someone requests versioned bucket support?

Reply:
We should create an issue, but probably won't start on it yet.